TSPL.docc/LanguageGuide/ErrorHandling.md

```diff
@@ -699,6 +699,204 @@ let photo = try! loadImage(atPath: "./Resources/John Appleseed.jpg")
+  they're always handled within that library.
+
+- In code that throws only errors that were thrown elsewhere,
+  like a function that takes a closure argument
+  and propagates any errors from that closure.
+  For a comparison between propagating a specific error type
+  and using `rethrows`,
+  see <doc:Declarations:Rethrowing-Functions-and-Methods>.
+
+For example,
+consider code that summarizes ratings
+and uses the following error type:
+
+```swift
+enum StatisticsError: Error {
+    case noRatings
+    case invalidRating(Int)
+}
+```
+
+To specify that a function throws only `StatisticsError` values as its errors,
+you write `throws(StatisticsError)` instead of only `throws`
+when declaring the function.
+This syntax is also called *typed throws*
+because you write the error type after `throws` in the declaration.
+For example,
+the function below throws `StatisticsError` values as its errors.
+
+```swift
+func summarize(_ ratings: [Int]) throws(StatisticsError) {
+    guard !ratings.isEmpty else { throw .noRatings }
+
+    var counts = [1: 0, 2: 0, 3: 0]
+    for rating in ratings {
+        guard rating > 0 && rating <= 3 else { throw .invalidRating(rating) }
+        counts[rating]! += 1
+    }
+
+    print("*", counts[1]!, "-- **", counts[2]!, "-- ***", counts[3]!)
+}
+```
+
+In the code above,
+the `summarize(_:)` function summarizes a list of ratings
+expressed on a scale of 1 to 3.
+This function throws an instance of `StatisticsError` if the input isn't valid.
+Both places in the code above that throw an error
+omit the type of the error
+because the function's error type is already defined.
+You can use the short form, `throw .noRatings`,
+instead of writing `throw StatisticsError.noRatings`
+when throwing an error in a function like this.
+
+When you write a specific error type at the start of the function,
+Swift checks that you don't throw any other errors.
+For example,
+if you tried to use `VendingMachineError` from examples earlier in this chapter
+in the `summarize(_:)` function above,
+that code would produce an error at compile time.
+
+You can call a function that uses typed throws
+from within a regular throwing function:
+
+```swift
+func someThrowingFunction() -> throws {
+    let ratings = [1, 2, 3, 2, 2, 1]
+    try summarize(ratings)
+}
+```
+
+The code above doesn't specify an error type for `someThrowingFunction()`,
+so it throws `any Error`.
+You could also write the error type explicitly as `throws(any Error)`;
+the code below is equivalent to the code above:
+
+```swift
+func someThrowingFunction() -> throws(any Error) {
+    let ratings = [1, 2, 3, 2, 2, 1]
+    try summarize(ratings)
+}
+```
+
+In this code,
+`someThrowingFunction()` propagates any errors that `summarize(_:)` throws.
+The errors from `summarize(_:)` are always `StatisticsError` values,
+which is also a valid error for `someThrowingFunction()` to throw.
+<!-- XXX Expand on subtyping here? -->
```

Comment:
I don't think you need to expand on subtyping. However, I would like to describe the `Never` case, because it's an important part of understanding how typed throws fits in the type system. For example, it could be something like this:

> Just like you can write a function that never returns with a return type of `Never`, you can write a function that never throws with `throws(Never)`:

    ```swift
    func nonThrowingFunction() throws(Never) {
      // ...
    }
    ```

> This function cannot throw because it is impossible to create a value of type `Never` to throw.